include/adc.h

```diff
@@ -294,6 +309,26 @@ struct adc_driver_api {
 #endif
 };
 
+/**
+ * @brief Retrieve the internal reference voltage
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @return The internal reference, in millivolts, or a negative error
+ * code such as `-ENOTSUP` where the reference is not supported.
+ */
+static inline int adc_ref_internal(struct device *dev)
+{
+	int rv = -ENOTSUP;
+
+	const struct adc_config_info *cip
+		= (const struct adc_config_info *)dev->config->config_info;
```

Comment:
Doesn't the `struct adc_config_info` need to be added to the other drivers that currently use `config_info` for their instance specific information, even if it's just populated with `-ENOTSUP`.  That is, wouldn't this currently return nonsense for them? (looks like mostly it would end up returning the base address of the instance registers from the current struct layouts).  Granted, it's a new API so nothing using them would currently fail, but it would be an odd result if something new tried to use it.